src/dos/cdrom_image.cpp

```diff
@@ -43,16 +47,30 @@ using namespace std;
 #define MAX_LINE_LENGTH 512
 #define MAX_FILENAME_LENGTH 256
 
+#ifdef DEBUG
+char* get_time() {
+	static char time_str[] = "00:00:00";
+	static time_t rawtime;
+	time(&rawtime);
+	const struct tm* ptime = localtime(&rawtime);
+	snprintf(time_str, sizeof(time_str), "%02d:%02d:%02d", ptime->tm_hour, ptime->tm_min, ptime->tm_sec);
+	return time_str;
+}
+#endif
+
 CDROM_Interface_Image::BinaryFile::BinaryFile(const char *filename, bool &error)
+	:TrackFile(RAW_SECTOR_SIZE)
 {
 	file = new ifstream(filename, ios::in | ios::binary);
-	error = (file == NULL) || (file->fail());
+	error = (file == nullptr) || (file->fail());
```

Comment:
*Just a comment, there's no need to change the code, I hope.*

Previously, I asked for usage of nullptr here, because I thought we might go for C++11 again. I am a bit surprised nullptr compiles with C++03 - the feature was added in C++11, but maybe nullptr itself existed already and only `nullptr_t` got added in new standard? Anyways, compilers do not seem to object to this, so maybe it's fine…

Reply:
From some reading, C++03 or c++0x was a long-time evolving standard the compilers tried to keep up with. You can see GCC's support for it evolve from early 4.x versions through to 4.8.1, here: https://gcc.gnu.org/projects/cxx-status.html#cxx11

When C++03 was finalized it was renamed C++11, and the C++03 standard was dropped/deprecated.  It just happens that if you use a modern-enough compiler that fully supported the final version of C++03 (such as gcc 4.8.1) then you wind up with full C++11 support as well :-).  Compilers newer than 4.8.1 and Clang 3.3 essentially given you full C++11 support when you ask for the c++0x standard.

If our CI workflow hit pre-4.8.1 compilers, then we might find one that doesn't support nullptr(_t).  Clang seems even more advanced.. we'd have to go back to pre-3.3 to exclude C++11 support!

Reply:
Ah, sorry, I should've read your changes more thoroughly and respond here before merging in - I have no problem with using C++11, so it gave me even less incentive to double-check what happened. And I forgot that C++0x brouhaha happened 10 years ago already and new generation of programmers might not know about it.

C++0x is not C++03! It is the next standard AFTER C++03. It was called 0x, because work on it started ~2005-2006 and committee expected it to be released in a year or two. 2008 came and went, then 2009 came and went, 2010 came and went and the standard was finally released in 2011. The compiler flag is preserved for backwards-compatibility. C++0x is "beta" version of C++11.

Reply:
Maybe we should flip over that configure.ac macro to just give us C++11? 

Maybe Qbix isn't aware of this detail when he mentioned 0x would be fine? (perhaps he thought it was to some prior early-2000's version?) Otherwise if he really is fine with beta-C++11 then why not just go with the finalized version..